crates/testlang/outputs/cargo/tests/src/query/parser_tests.rs

```diff
@@ -52,12 +52,60 @@ fn test_zero_or_more_canonicalisation() {
 // Test the error message on parse failure
 #[test]
 fn test_parsing_error() {
-    let result = Query::parse(r#"@root [_ ..."#);
+    let result = Query::parse(r#"@root [_"#);
     match result {
-        Ok(_) => panic!("Expected error"),
+        Ok(_) => panic!("Expected parse failure"),
+        Err(e) => assert_eq!(e.message, "Parse error:\nexpected ']' at: \nAlt at: [_\n"),
+    }
+}
+
+#[test]
+fn test_fails_parsing_ellipsis() {
+    let result = Query::parse(r#"[_ ...]"#);
+    match result {
+        Ok(_) => panic!("Expected parse failure"),
+        Err(e) => assert_eq!(
+            e.message,
+            "Parse error:\nFail at: ...]\nin section 'deprecated ellipsis operator', at: ...]\n"
```

Comment:
I noticed these tests don't check for error location. I suggest adding that to the assertion.
I recently filed an issue that it is not reported correctly: #1041
Maybe that is impacted/fixed by the newly introduced  `recognize_as_failure()`.

Reply:
Location tracking is not implemented at all for query parsing. The `QueryError` is constructed [here](https://github.com/manastech/slang/blob/feat%2Fquery-syntax-changes/crates/metaslang/cst/src/query/parser.rs#L46).

I can work on this (and improve on the error message issue mentioned above) but I'd prefer to do it in a separate PR.

Reply:
Sounds good.